is_core/generic_views/inlines/inline_form_views.py

```diff
@@ -96,21 +109,27 @@ def get_initial(self):
         return self.initial[:]
 
     def get_can_delete(self):
-        return self.can_delete and not self.readonly
+        return (
+            (self.get_can_add() or (self.can_delete and self.permission.has_permission('delete', self.request, self)))
+            and not self.readonly
+        )
```

Comment:
newly added not saved row can be deleted too. Therefore delete field must be generated although stored instances cannot be deleted.